sql/src/test/java/io/crate/testing/SQLTransportExecutor.java

```diff
@@ -175,6 +190,40 @@ private boolean isJdbcEnabled() {
         return false;
     }
 
+    /**
+     * @return true if a class or method in the stacktrace contains a @UseSemiJoins annotation
+     * and based on the ration provided
+     * <p>
+     * Method annotations have higher priority than class annotations.
+     */
+    private boolean isSemiJoinsEnabled() {
+        StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
+        for (StackTraceElement element : stackTrace) {
```

Comment:
Is there a nicer way than to go loop through the stracktrace? We should have the test class available somehow during tests.

Reply:
Don't think so, I've searched a bit.
Take a look here: https://stackoverflow.com/questions/421280/how-do-i-find-the-caller-of-a-method-using-stacktrace-or-reflection
With Java9 though ;) http://openjdk.java.net/jeps/259

Reply:
Perhaps this could help to avoid loop through the stack: https://stackoverflow.com/questions/473401/get-name-of-currently-executing-test-in-junit-4